### DISCUSSION TITLE
为什么发布的模型维度是 3584，而不是 2 的幂？

### DISCUSSION THREAD

**Discussion by silencelamb** [QwenLM/Qwen3#1036] [General]
您好，非常感谢开源了如此优秀的 Qwen 系列模型！

我注意到，发布的模型使用了 3584 这一维度。这个数是 7 的倍数，但并不属于 2 的幂，这与许多模型常见的维度选择（如 4096 等）有所不同。据我了解，通常 2 的幂shape的矩阵乘法在加速硬件上更容易开发实现高性能计算的kernel。请问选择 3584 作为模型维度的原因是什么？是否有特定的架构设计或性能优化方面的考量？

期待您的解答，再次感谢！

**Comment by jklj077** [comment]
3584是7B的hidden size，其它模型hidden size各有不同。

> 通常 2 的幂shape的矩阵乘法在加速硬件上更容易开发实现高性能计算的kernel

也不完全是。考虑NVIDIA硬件，使用tensor core，大多数时候半精度计算维度是8倍数就okay了；GEMM考虑tile size的话基本都是256x128/128x256的时候效率最好。维度不必完全是2的幂。

> 请问选择 3584 作为模型维度的原因是什么？是否有特定的架构设计或性能优化方面的考量？

综合考量，效果为主，维度尽量被512整除（考虑tensor parallel以及量化的block大小）。
